src/DIRAC/WorkloadManagementSystem/Utilities/PilotWrapper.py

```diff
@@ -106,6 +111,20 @@ def pilotWrapperScript(
     if envVariables is None:
         envVariables = {}
 
+    if CVMFS_locations:
+        # Here we are making the assumption that, if CVMFS_locations is, e.g., ['/cvmfs/somewhere', '/cvmfs/elsewhere']
+        # and the project is 'LHCb',
+        # then the pilot can maybe be found at locations
+        # - file:/cvmfs/somewhere/lhcbdirac/pilot
+        # - file:/cvmfs/elsewhere/lhcbdirac/pilot
+        project = "dirac"
+        if "-l" in pilotOptions:
+            project = pilotOptions.split(" ")[pilotOptions.split(" ").index("-l") + 1].lower() + "dirac"
```

Comment:
For this specific case yes, it would make the code more readable, but then it would need to be anyway converted back to a string.

Reply:
Well, the `SiteDirector` would create a `dict` and then the `PilotWrapper` would convert it into a string.
But indeed, this would be only for this single use case.